addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/handler/AbstractMilightBridgeHandler.java

```diff
@@ -0,0 +1,178 @@
+    }
+
+    /**
+     * Creates a connection and other supportive objects.
+     *
+     * @param addr
+     */
+    protected abstract void startConnectAndKeepAlive(InetAddress addr);
+
+    protected abstract Runnable getKeepAliveRunnable();
+
+    private void connectAndKeepAlive() {
+        Object host_config_obj = thing.getConfiguration().get(MilightBindingConstants.CONFIG_HOST_NAME);
+        String host_config = ((host_config_obj instanceof String) ? (String) host_config_obj
+                : (host_config_obj instanceof InetAddress) ? ((InetAddress) host_config_obj).getHostAddress() : null);
+
+        InetAddress addr = null;
+
+        try {
+            addr = InetAddress.getByName(host_config);
+        } catch (UnknownHostException ignored) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No address known!");
+            return;
+        }
+
+        bridgeid = (String) thing.getConfiguration().get(MilightBindingConstants.CONFIG_ID);
+
+        // Version 1/2 do not support response messages / detection. We therefore directly call bridgeDetected(addr).
+        if (bridgeid == null || bridgeid.length() != 12) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridgeID of length 12!");
+            return;
+        }
+
+        startConnectAndKeepAlive(addr);
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        this.thing = thing;
+        if (com == null) {
+            return;
+        }
+
+        // Create a new communication object if the user changed the IP configuration.
+        Object host_config_obj = thing.getConfiguration().get(MilightBindingConstants.CONFIG_HOST_NAME);
+        String host_config = ((host_config_obj instanceof String) ? (String) host_config_obj
+                : (host_config_obj instanceof InetAddress) ? ((InetAddress) host_config_obj).getHostAddress() : null);
+        if (host_config != null && !host_config.equals(com.getAddr().getHostAddress())) {
+            try {
+                com.setAddress(InetAddress.getByName(host_config));
+            } catch (UnknownHostException e) {
+                e.printStackTrace();
+            }
+        }
+
+        // Create a new communication object if the user changed the port configuration.
+        BigDecimal port_config = (BigDecimal) thing.getConfiguration().get(MilightBindingConstants.CONFIG_CUSTOM_PORT);
+        if (port_config != null && port_config.intValue() > 0 && port_config.intValue() <= 65000
+                && port_config.intValue() != com.getPort()) {
+            com.setPort(port_config.intValue());
+        }
+
+        // Create a new communication object if the user changed the bridge ID configuration.
+        String id_config = (String) thing.getConfiguration().get(MilightBindingConstants.CONFIG_ID);
+        if (id_config != null && !id_config.equals(bridgeid)) {
+            connectAndKeepAlive();
```

Comment:
Yes, this binding was initially implemented in a way that **initialize()** is expected to be called only once and therefore only one sendqueue exists. All bulbs are referencing this very first object and never get updated with another. Additionally I didn't know that there is a **handleConfigurationUpdate()** method that needs to be overwritten which otherwise calls **initialize()** again and triggers the aforementioned problem.